oksdk.js

```diff
@@ -554,7 +554,17 @@
         if (state.sessionKey) {
             query += '&st.session_key=' + state.sessionKey;
         }
-        window.open(query);
+
+        let w = window.open(query);
+        if (w === null) {
+            let event = {
+                "data": {
+                    "code": "popup_blocked"
+                }
+            };
+
+            window.postMessage(JSON.stringify(event), '*');
```

Comment:
Because original code doesn't return anything, and because it's callers doesn't handle any result of widgetOpen().

Reply:
callers can be made aware of the return value within same pull request

postmessage is designed for async responses; here we have a sync response, so i think it's better to return sync result instantly
especially because as googling shown - sometimes you can get null while the window will be opened, so it should be upon the developer what to do, not reported as a generic widget error unlike other return values via postmessage

Reply:
My hotfix just works and allows an application to be informed using the default manner.
But you can fix this problem in any way ;)

Did you remember that widgetOpen() is internal SDK function, that called within SDK only?

Reply:
what's the problem changing the places it's called from windowOpen(...) to return windowOpen(...) in addition to the return window !== null  in windowOpen itself?

Reply:
ps: let cannot be used, since it is not supported in older browsers, including webview on many 4.x android devices

Reply:
Internal function widgetOpen() called within various of public functions, such as widgetMediatopicPost(), widgetInvite(), widgetSuggest(). We can handle window.open() result within each of these functions, or we can handler result in one function. I prefer doing one common thing in one function, and not propagate it over all code.

And I just suggest hotfix, that allows handling error. As I mentioned before, you can fix this problem in any way ;)

Reply:
ok, i will do it